tools/vuln_valid/index.js

```diff
@@ -57,23 +56,29 @@ const npmModel = joi.object().keys({
   coordinating_vendor: joi.string().allow(null).required()
 });
 
+function validateVuln(filePath, model){
+  const vuln = JSON.parse(fs.readFileSync(filePath));
+  const result = joi.validate(vuln, model);
+  if (result.error) {
+    console.error(result.error)
+    throw result.error;
+  }
+}
+
 function validate(dir, model) {
   fs.readdirSync(dir)
     .forEach((name) => {
       const filePath = path.join(dir, name);
-      try {
-        const vuln = JSON.parse(fs.readFileSync(filePath));
-        const result = joi.validate(vuln, model);
-        if (result.error) {
-          throw result.error;
-        }
-      } catch (err) {
-        console.log(`File ${filePath}:`);
-        console.log(err);
-        process.exitCode = 1;
-      }
+      validateVuln(filePath, model)
     });
 }
 
 validate(vulnPaths.core, coreModel);
 validate(vulnPaths.npm, npmModel);
+
+module.exports = {
```

Comment:
you can't just export the functionality because we use this file to actually run the validation (see the calls above. and every time its required it will actually run them.

thus, I suggest another level of refactoring:
1. keep these exports as you added them so they can be consumed for tests
2. remove the two lines above of the validate() calls and push them into another file - and we will call that other file from the npm run-script to run the validation

it would actually be great if this index.js file be the top-level code that calls validate() and you move all of the logic in this index.js file to vulnValidate.js instead (or some other similar name).

Reply:
Okay, makes sense. Will do the necessary.